corgi/tasks/errata_tool.py

```diff
@@ -130,8 +135,18 @@ def update_variant_repos() -> None:
 
             pv_channels = []
             for repo in et_variant_data["repos"]:
-                repo, created = Channel.objects.get_or_create(
-                    name=repo, defaults={"type": Channel.Type.CDN_REPO}
+                # Filter out inactive repos in pulp and get content_set
+                try:
+                    rpm_repo = CollectorRPMRepository.objects.get(name=repo)
+                except CollectorRPMRepository.DoesNotExist:
+                    logger.debug("Not creating Channel for inactive repo %s", repo)
+                    continue
+                repo, created = Channel.objects.update_or_create(
+                    name=repo,
+                    defaults={
+                        "type": Channel.Type.CDN_REPO,
+                        "meta_attr": {"content_set": rpm_repo.content_set},
```

Comment:
This looks like a bug. rpm_repo.content_set can (I assume) change from time to time, but because it's inside the defaults= dict, the content_set will only be stored when the Channel object is first created.

Whenever we try to update the Channel object, we'll only update the keyword arguments (name, which I assume won't change) and everything else in the defaults dict will stay unchanged.

Reply:
That's not how I understand update_or_create. Defaults are used to update the fetched object, which is what we want in this case. See https://docs.djangoproject.com/en/3.2/ref/models/querysets/#update-or-create it says "The defaults is a dictionary of (field, value) pairs used to update the object.". If the conent_set is updated we want to set that on the named repo. I don't think this will ever happen, but if it does the code here should update the conent_set field using the value passed in the defaults dict.

Inactive in this case mean it has RPMs shipped to it. I don't think there is a case when a Repo would have RPMs removed from it, so it won't become inactive.

Reply:
Whoops, you're right - I was misremembering the behavior as the other way around (only update kwargs). And re: inactive, sounds good - thanks for explaining.